crates/ruff_linter/src/rules/ruff/rules/dict_update_single.rs

```diff
@@ -0,0 +1,127 @@
+use ruff_diagnostics::{AlwaysFixableViolation, Diagnostic, Edit, Fix};
+use ruff_macros::{derive_message_formats, violation};
+use ruff_python_ast::{DictItem, Expr, ExprAttribute, ExprCall, ExprDict};
+use ruff_python_semantic::analyze::typing;
+use ruff_python_semantic::SemanticModel;
+use ruff_text_size::Ranged;
+
+use crate::checkers::ast::Checker;
+
+/// Real type: [`Expr::Name`]
+type DictRef = Expr;
+type Argument = Expr;
+type Key = Expr;
+type Value = Expr;
+
+/// ## What it does
+/// Checks for `dictionary.update({single: item})`.
+///
+/// ## Why is this bad?
+/// Calling `dict.update()` with a single-item dictionary
+/// is equivalent to setting the item directly,
+/// but more verbose and potentially more expensive.
+///
+/// ## Example
+///
+/// ```python
+/// def foo(d: dict[str, int]) -> None:
+///     d.update({"lorem": 42})
+/// ```
+///
+/// Use instead:
+///
+/// ```python
+/// def foo(d: dict[str, int]) -> None:
+///     d["lorem"] = 42
+/// ```
+#[violation]
+pub struct DictUpdateSingle;
+
+impl AlwaysFixableViolation for DictUpdateSingle {
+    #[derive_message_formats]
+    fn message(&self) -> String {
+        "Literal dictionary with single item in `dict.update()` call".to_string()
```

Comment:
It's unclear to me what *Literal dictionary* means in this context. 
```suggestion
        "`dict.update` with single dictionary argument".to_string()
```